haplotypecaller-gvcf-gatk3.wdl

```diff
@@ -48,112 +70,130 @@ workflow HaplotypeCallerGvcf_GATK3 {
         input_bam = input_bam,
         input_bam_index = input_bam_index,
         interval_list = interval_file,
-        gvcf_name = gvcf_name,
-        gvcf_index = gvcf_index,
+        output_filename = output_filename,
         ref_dict = ref_dict,
         ref_fasta = ref_fasta,
-        ref_fasta_index = ref_fasta_index
+        ref_fasta_index = ref_fasta_index,
+        hc_scatter = hc_divisor,
+        make_gvcf = making_gvcf,
+        docker = gatk_docker,
+        gatk_path = gatk_path
     }
   }
 
   # Merge per-interval GVCFs
   call MergeVCFs {
     input:
-      input_vcfs = HaplotypeCaller.output_gvcf,
-      input_vcfs_indices = HaplotypeCaller.output_gvcf_index,
-      vcf_name = gvcf_name,
-      vcf_index = gvcf_index
+      input_vcfs = HaplotypeCaller.output_vcf,
+      input_vcfs_indexes = HaplotypeCaller.output_vcf_index,
+      output_filename = output_filename,
+      docker = picard_docker,
+      picard_path = picard_path
   }
 
   # Outputs that will be retained when execution is complete
   output {
-    File output_merged_gvcf = MergeVCFs.output_vcf
-    File output_merged_gvcf_index = MergeVCFs.output_vcf_index
+    File output_vcf = MergeVCFs.output_vcf
+    File output_vcf_index = MergeVCFs.output_vcf_index
   }
 }
 
 # TASK DEFINITIONS
 
 # HaplotypeCaller per-sample in GVCF mode
 task HaplotypeCaller {
+  
+  # Command Paramters
   File input_bam
   File input_bam_index
-  String gvcf_name
-  String gvcf_index
+  String output_filename
   File ref_dict
   File ref_fasta
   File ref_fasta_index
   File interval_list
   Int? interval_padding
   Int? contamination
   Int? max_alt_alleles
+  Boolean make_gvcf
+  Int hc_scatter
 
-  Int preemptible_tries
-  Int disk_size
-  String mem_size
-
-  String docker_image
   String gatk_path
   String java_opt
 
+  # Runtime parameters
+  String docker
+  Int? machine_mem_gb
+  Int? disk_space_gb
+  Boolean use_ssd = false
+  Int? preemptible_attempts
+
+  Float ref_size = size(ref_fasta, "GB") + size(ref_fasta_index, "GB") + size(ref_dict, "GB")
+  Int disk_size = ceil(((size(input_bam, "GB") + 30) / hc_scatter) + ref_size) + 20
+
   command {
-    java ${java_opt} -jar ${gatk_path}GenomeAnalysisTk.jar \
+    java ${java_opt} -jar ${gatk_path}GenomeAnalysisTK.jar \
       -T HaplotypeCaller \
       -R ${ref_fasta} \
       -I ${input_bam} \
-      -o ${gvcf_name} \
+      -o ${output_filename} \
       -L ${interval_list} \
       -ip ${default=100 interval_padding} \
```

Comment:
Where did this come from? It's not in the production version: https://github.com/broadinstitute/dsde-pipelines/blob/master/genomes_in_the_cloud/single_sample/PairedSingleSampleWf.wdl This could throw errors.  If you have adjacent intervals in your input intervals then there could be repeated variants between shard x and shard x+1 that will lead to non-decreasing positions (and duplicates) when the shards are merged.  HaplotypeCaller has default 100bp padding on the assembly region and a default minimum of 50bp assembly region size anyway.

Reply:
Not sure. 
I'll switch over to the provided commands and to the gitc docker. 
Thanks

Reply:
The GATK3 docker you have should be fine and your command just needs the -ip line to be dropped. It's better to use the max_alternate_alleles from production too because that will speed joint calling up for WGS in gross indel regions.

variant_index_parameter and variant_index_type are redundant provided your GVCF has a .g.vcf (or .g.vcf.gz) extension.  It's on my list to update production.

Reply:
I'll remove the redundant parameters and set the max_alt to 3. 
Since I don't know where the original command came from its probably a good idea to use the one from production. It also helps keep things consistent so i'll use the docker and commands from production.